connector/client/client.go

```diff
@@ -78,42 +93,51 @@ func (client ConnectorClient) Post(target url.URL, activity schema.Activity) err
 
 func (client *ConnectorClient) getToken() (string, error) {
 
-	data := url.Values{}
-	data.Set("grant_type", "client_credentials")
-	data.Set("client_id", client.Credentials.GetAppID())
-	data.Set("client_secret", client.Credentials.GetAppPassword())
-	data.Set("scope", auth.ToChannelFromBotOauthScope)
+	if cache == nil || cache.IsExpired() {
```

Comment:
It's little difficult to understand. Let's change this condition `if !cache.IsExpired()` and return `cache.token` here